### ISSUE TITLE
ReadStream keeps reading after destroy was called.

### ISSUE THREAD

**Issue by hutorny** [nodejs/node-v0.x-archive#7052]
Hello

In the code snippet below data are read from a character device (slow endless stream).
The _reader.read_ reads a portion of the stream and returns true when 'end of portion' condition met. The code snippet then destroys the stream.
Nevertheless, the stream keeps reading until the condition met again, e.g. reads another portion and only then stops, 

```
    fs.createReadStream(name)
        .on('data', function(chunk) {
            console.log(chunk.length);
            if( reader.read(chunk) ) {
                res.send(200, reader.data);
                console.log('Destroying');
                this.destroy();
            }
    }); 
```

The log has the following lines

```
432
GET /test 200 2060ms - 2kb
Destroying
432
Destroying
```

**Comment by jasnell** [comment]
@joyent/node-coreteam @trevnorris ... the reported behavior is still happening in v0.10.39-pre but not in v0.12 or io.js. Do we need to fix in v0.10?

**Comment by trevnorris** [comment]
I can't remember the v0.10 streams implementation. if the fix is simple then sure. if not then I wouldn't worry about it.
